File: client/src/app/entities/item.ts
import {User} from "./user";
import {Itemstatus} from "./itemstatus";
import {Category} from "./category";
import {DataPage} from "../shared/data-page";
import {Supplier} from "./supplier";

export class Item {
  id: number;
  code: string;
  tocreation: string;
  description: string;
  creator: User;
  name: string;
  photo: string;
  qty: number;
  rop: number;
  price: number;
  category: Category;
  itemstatus: Itemstatus;

  supplierList: Supplier[];


  constructor(id: number=null) {
    this.id = id;
  }
}
export class ItemDataPage extends DataPage{
  content: Item[];
}

File: server/src/main/java/bit/project/server/entity/Item.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDateTime tocreation;

    @Lob
    private String description;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;

    private String name;
    private String photo;
    private Integer qty;
    private Integer rop;
    private BigDecimal price;

    @ManyToOne
    private Category category;

    @ManyToOne
    private Itemstatus itemstatus;


    @ManyToMany
    @JoinTable(
            name = "itemsupplier",
            joinColumns = @JoinColumn(name = "item_id" ,referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "supplier_id" , referencedColumnName = "id")
    )
    private List<Supplier> supplierList;

    @JsonIgnore
    @OneToMany(mappedBy = "item")
    private List<Purchaseitem> purchaseitemList;

    @JsonIgnore
    @OneToMany(mappedBy = "item")
    private List<Saleitem> saleitemList;

    public Item(Integer id) {
        this.id = id;
    }

    public Item(Integer id, String code, String name) {
        this.id = id;
        this.code = code;
        this.name = name;
    }
}
